zql/Makefile

```diff
@@ -0,0 +1,17 @@
+GOIMPORTS = goimports
+PEGJS = pegjs
+PIGEON = pigeon
+
+all: zql.go zql.js
+
+run: all
+	go run ./main
+
+.PHONY: zql.go
+zql.go:
+	cpp -DGO -E -P zql.peg | $(PIGEON) -o $@
+	$(GOIMPORTS) -w $@
+
+.PHONY: zql.js
+zql.js:
+	cpp -E -P zql.peg | $(PEGJS) -o $@
```

Comment:
shouldn't this all be in the top level Makefile?
also, i don't have any experience with it, maybe its not suitable, but was `go generate` for this step ever considered?  it wouldn't handle the javascript of course...

Reply:
> shouldn't this all be in the top level Makefile?

I put these rules here since they're only used when modifying zql.peg, and keeping related things close seemed tidy. Fine with me if you want to move it, though.

> also, i don't have any experience with it, maybe its not suitable, but was go generate for this step ever considered? it wouldn't handle the javascript of course...

I don't have experience beyond reading about it, but I don't see any benefit to using it here.